book/src/development/trait_checking.md

```diff
@@ -94,6 +94,33 @@ impl LateLintPass<'_> for CheckTokioAsyncReadExtTrait {
 }
 ```
 
+## Creating traits programmatically
+
+Traits are often generic over a type e.g. `Borrow<T>` is generic over `T`, and rust allows us to implement a trait for
+a specific type. For example, we can implement `Borrow<str>` for a hypothetical type `Foo`. Let's suppose that we
+would like to find whether our type actually implements `Borrow<[u8]>`. To do so, we need to supply a type that
+represents `[u8]`, but `[u8]` is also a generic, it's a slice over `u8`. We can create this type using Ty::new_slice
+method. The following code demonstrates how to do this:
+
+```rust
+
+use rustc_middle::ty::Ty;
+use clippy_utils::ty::implements_trait;
+use rustc_span::symbol::sym;
+
+let ty = todo!();
+let borrow_id = cx.tcx.get_diagnostic_item(sym::Borrow).unwrap(); // avoid unwrap in real code
+if implements_trait(cx, ty, borrow_id, &[Ty::new_slice(cx.tcx, cx.tcx.types.u8).into()]) {
```

Comment:
I would place the `Ty::new_slice` in its own line before the `if`, as this is what this documentation is about. Then maybe add a comment like

```rust
let slice = Ty::new_slice(...) // <- creating the type here
```